doc/source/dataset.rst

```diff
@@ -0,0 +1,50 @@
+Ray Datasets
+============
+
+.. warning::
+  This feature is currently in preview, and its API may change.
+
+Ray Datasets are the standard way to load and exchange data between Ray libraries and applications. You can think of a Ray Dataset as a distributed collection of data records, which can be either serializable Python objects or structured data (Arrow records). Datasets support basic parallel transformations, and can be ``split()`` into pieces for ingest into Ray's distributed machine learning libraries.
+
+For those familiar with Apache Arrow, Ray Datasets can be thought of as implementing "distributed Arrow".
+
+Under the hood, Datasets are implemented using Ray objects. Data records are organized into data *blocks*, each of which is a Ray object holding an Arrow table. A Dataset is simply a list of these block references and their metadata. Datasets implement parallel operations using Ray tasks and actors to transform their data blocks. Since Datasets are just lists of Ray object refs, they can also be passed between Ray tasks and actors just like any other object. Datasets support conversion to/from several more featureful dataframe libraries (e.g., Spark, Dask, Modin, MARS), and also conversion into TensorFlow and PyTorch dataset formats.
+
```

Comment:
All I'm really looking for is

1. matrix of supported (or to be supported) inputs / outputs 
2. High level summary APIs that I can use

Reply:
+1